File: OrchidCore/src/main/java/com/eden/orchid/impl/tasks/QuitCommand.java
package com.eden.orchid.impl.tasks;

import com.eden.orchid.Orchid;
import com.eden.orchid.api.OrchidContext;
import com.eden.orchid.api.options.annotations.Description;
import com.eden.orchid.api.tasks.OrchidCommand;
import com.google.inject.Provider;

import javax.inject.Inject;

@Description("Run the main Orchid build process.")
public final class QuitCommand extends OrchidCommand {

    private final Provider<OrchidContext> contextProvider;

    @Inject
    public QuitCommand(Provider<OrchidContext> contextProvider) {
        super(1000, "quit");
        this.contextProvider = contextProvider;
    }

    @Override
    public String[] parameters() {
        return new String[0];
    }

    @Override
    public void run(String commandName) {
        contextProvider.get().broadcast(Orchid.Lifecycle.EndSession.fire(this));
    }
}

File: OrchidCore/src/main/kotlin/com/eden/orchid/impl/commands/QuitCommand.kt
package com.eden.orchid.impl.commands

import com.eden.orchid.Orchid
import com.eden.orchid.api.OrchidContext
import com.eden.orchid.api.options.annotations.Description
import com.eden.orchid.api.tasks.OrchidCommand

@Description("Run the main Orchid build process.")
class QuitCommand : OrchidCommand(1000, "quit") {

    override fun parameters(): Array<String> {
        return emptyArray()
    }

    override fun run(context: OrchidContext, commandName: String) {
        context.broadcast(Orchid.Lifecycle.EndSession.fire(this))
    }
}

